### PR TITLE
Added AllBooks and ViewModel

### PR DESCRIPTION
**Pull request by rashigoel**

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -16,6 +16,10 @@ android {
             proguardFiles getDefaultProguardFile('proguard-android-optimize.txt'), 'proguard-rules.pro'
         }
     }
+    compileOptions {
```
Comment (prabhavgupta) [inline app/build.gradle]: Purpose?

File: app/src/main/java/com/street35/booked/MainActivity.java [hunk 2]
```diff
@@ -2,28 +2,39 @@
...
+import androidx.fragment.app.FragmentManager;
+
 import android.view.MenuItem;
 import android.widget.TextView;
 
 public class MainActivity extends AppCompatActivity {
     private TextView mTextMessage;
-
+    AllBooks allBooksFragment = new AllBooks();
+    FragmentManager fm = getSupportFragmentManager();
```
Comment (prabhavgupta) [inline app/src/main/java/com/street35/booked/MainActivity.java]: Avoid declaring unused variables.

File: app/src/main/java/com/street35/booked/ui/AllBooks.java [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+import com.street35.booked.R;
+import com.street35.booked.viewmodel.AllBooksViewModel;
+
+public class AllBooks extends Fragment {
+
+    private AllBooksViewModel mViewModel;
+
+    public static AllBooks newInstance() {
+        Log.d("ERROR", "Oppppsssieeeeeeeeeeeeeeeeeeeeee");
```
Comment (prabhavgupta) [inline app/src/main/java/com/street35/booked/ui/AllBooks.java]: Prefer to write meaningful logs and avoid pushing it to the master repo.

File: app/src/main/java/com/street35/booked/ui/BooksAdapter.java [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+public class BooksAdapter extends ArrayAdapter<Book> {
+
+    private Context bookContext;
+    private List<Book> bookList = new ArrayList<>();
+
+    public BooksAdapter(@NonNull Context context, int resource, @NonNull List<Book> objects) {
+        super(context, resource, objects);
+        context = bookContext;
+        objects = bookList;
```
Comment (prabhavgupta) [inline app/src/main/java/com/street35/booked/ui/BooksAdapter.java]: Confirm once

File: app/src/main/res/layout/all_books_fragment.xml [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+<FrameLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    tools:context="com.street35.booked.ui.AllBooks">
+
+    <TextView
+        android:layout_width="match_parent"
+        android:layout_height="match_parent"
+        android:text="All Books" />
```
Comment (prabhavgupta) [inline app/src/main/res/layout/all_books_fragment.xml]: Use strings.xml for strings

File: app/build.gradle (hunk 1 above)
Comment (rashigoel) [reply app/build.gradle]: To use JAVA 8 features.

File: app/src/main/res/layout/all_books_fragment.xml (hunk 5 above)
Comment (rashigoel) [reply app/src/main/res/layout/all_books_fragment.xml]: For testing purpose only.

**Pull request merged**